lib/Basic/Platform.cpp

```diff
@@ -118,6 +118,12 @@ StringRef swift::getPlatformNameForTriple(const llvm::Triple &triple) {
   case llvm::Triple::FreeBSD:
     return "freebsd";
   case llvm::Triple::Win32:
+    if (triple.isWindowsCygwinEnvironment())
+      return "cygwin";
+
+    if (triple.isWindowsGNUEnvironment())
+      return "mingw";
+
```

Comment:
Yes, a nested switch.  You should be able to do a covered switch there, and have most of the environments be unreachable.